hudi-common/src/main/java/org/apache/hudi/common/model/TableConfig.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+/*
+Represents object with all the topic level overrides for multi table delta streamer execution
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
any reason this is in the `hudi-common` package?

Reply:
Basically it is a POJO and hudi-common is having most of the POJOs. Also since hudi-common is added as a dependency in most of the modules, it will be easier this way to use this class anywhere in future if need be. :)

Reply:
Get your point, will move the class to utilities package. Can you please elaborate what do you mean by TargetTableConfig. All the properties in this class only apply for Source. I guess with few changes in validations part, the same class can be used for non-kafka sources too? Let me know your thoughts on this. @bvaradar

Reply:
In TableConfig.java, there are kafka related configs and there are configs like Keygeneratorclass. which are related to upstream source whereas configs related to hive-sync and others apply to the written target dataset.  My understanding is TableConfig is coming from DFSProperties. IMO, decoupling source and table config separately will be cleaner. Let me know if there is any implementation difficulties around it

Reply:
@bvaradar If we assume TableConfig is coming from DFSProperties (I am doing that change anyways :) ), then what extra benefit are we getting by decoupling source and target configs?  Because implementation wise I will be merging source and target configs into single TypedProperties instance after reading them separately since everywhere we are passing only single TypedProperties instance for reading relevant configs. 

If TableConfig is read as DFSProperties instance (key-value pair), then non-kafka sources are automatically handled. I understand maintaining source and target configs separately is cleaner, but then it will be an extra overhead for users to maintain 2 separate files. 

Please help me understand.

Reply:
For example, for DFS source we need to specify hoodie.deltastreamer.source.dfs.root property. This can be specified in corresponding table config coming as DFSProperties instance. 

Again I am assuming in a single installation of MultiTableDeltaStreamer, all the tables will use same source (which I believe is a valid assumption to make). Let me know your thoughts on this.

Reply:
@pratyakshsharma : Sounds fair. Let's proceed with having TableConfig containing both Source and Sink config but rename the class to reflect it. The name TableConfig seems misleading to me.  I will go through the change once you address other comments.

Regarding your comment about having same source, I think you meant source type (Kafka, DFS,...).  As the current TableConfig takes care of 1 pair of Source<->Sink configs (with configs in separate folders) anyways, there is no need to force that restriction. Let me know if I am missing something ?

Reply:
@bvaradar yes since we are going ahead with DFSPropertiesConfiguration, I am going to remove TableConfig class completely. 

Regarding comment having same source, yes I meant same source type. But since source class gets passed as command line parameter (--source-class), it is better to put this restriction. Because we cannot override command line parameter in source <-> sink config key value pairs. If you really want to have this flexibility of overriding source type also for every pair, I will have to define one new property in DataSourceWriteOptions and use it to configure source class.

Reply:
Sounds good on removing TableConfig. Since, HoodieMultiTableDeltaStreamer is going to be a new main class, is it possible to avoid source-class to be a command-line-parameter and instead be part of the DFSProperties ? Anyways, lets not get too bogged down by it for now. Once you have the changes ready, will do another pass. If minor, we can try to do those as follow-ups.